unlock-app/src/actions/fullScreenModals.ts

```diff
@@ -5,9 +5,10 @@ export const DISMISS_MODAL = 'fullScreenModal/DISMISS'
 
 // General events, in terms of which other kinds of modal interaction should be
 // defined.
-export const launchModal = (kindOfModal: KindOfModal) => ({
+export const launchModal = (kindOfModal: KindOfModal, data?: any) => ({
```

Comment:
The previous modals were gettin their data from actions. do you think we could keep that approach for consistency?

Reply:
The other ones always rendered the same thing no matter what, and communicated the bundles through the `dispatch` prop passed in to the modal template (i.e., no callback, just putting the credential payload on the redux chain). This one is a bit different since the template itself needs the value for the QR code.

I agree that my approach here is a little unclean. It's because the original modal infrastructure I implemented here was to have many kinds of static modals. Since right now in practice we have only 2 kinds, it might be better to have distinct actions instead of one parent action with different kinds of modals attached.